pkg/kademlia/kademlia.go

```diff
@@ -330,7 +337,7 @@ func (k *Kad) announce(ctx context.Context, peer swarm.Address) error {
 
 	err := k.discovery.BroadcastPeers(ctx, peer, addrs...)
 	if err != nil {
-		_ = k.p2p.Disconnect(peer)
+		// _ = k.p2p.Disconnect(peer)
```

Comment:
why?

Reply:
A couple of things can go wrong here (namely in sendPeers)
stream, err := s.streamer.NewStream(ctx, peer, nil, protocolName, protocolVersion, peersStreamName)
addr, err := s.addressBook.Get(p)
if err := w.WriteMsg(&peersRequest); err != nil 

What I was experiencing was losing almost every connection within a couple of minutes of establishment, so I just commented this out so that I potentially don't lose so many connections